File: PASCAL/Cap08/c08ex04.pas
program c08ex04;

var
  Nomes: array[1..5] of String;
  X: String;
  I, J: Integer;

begin
  WriteLn('CLASSIFICACAO DE NOMES (DECRESCENTE)');
  WriteLn;

  for I := 1 to 5 do
    begin
      Write('Entre o ', I, 'o. nome: ');
      ReadLn(Nomes[I]);
    end;

  for I := 1 to 4 do
    for J := I + 1 to 5 do
      if (Nomes[I] < Nomes[J]) then
        begin
          X := Nomes[I];
          Nomes[I] := Nomes[J];
          Nomes[J] := X;
        end;

  WriteLn;
  for I := 1 to 5 do
    WriteLn(I, 'o. nome: ', Nomes[I]);

  WriteLn;
  Write('Tecle <Enter> para encerrar... ');
  ReadLn;
end.

File: TYPESCRIPT/Cap08/c08ex04.ts
// c08ex04.ts

import * as rls from 'readline-sync';

let Nomes: string[] = new Array(5).fill('').map(() => 
new Array(100).fill('').join(''));
let X: string = new Array(100).fill('').join('');
let I: number, J: number;

console.log("CLASSIFICACAO DE NOMES (DECRESCENTE)\n");

for (I = 0; I <= 4; ++I) {
  Nomes[I] = rls.question(`Entre o ${I + 1}o. nome: `);
}

for (I = 0; I <= 3; ++I) {
  for (J = I + 1; J <= 4; J++) {
    if (Nomes[I] < Nomes[J]) {
      X = Nomes[I];
      Nomes[I] = Nomes[J];
      Nomes[J] = X;
    }
  }
}

console.log("");
for (I = 0; I <= 4; ++I) {
  console.log(`${I + 1}o. nome: ${Nomes[I]}`);
}

console.log("");
process.stdout.write("Tecle <Enter> para encerrar... ");
rls.question();
